### DISCUSSION TITLE
Public Roadmap?

### DISCUSSION THREAD

**Discussion by osdiab** [remix-run/remix#4162] [General]
Is there a public roadmap for what the Remix team is focusing on somewhere? It would be great to know whether or not features that are currently missing that may prove challenging to my team if we switch to Remix are actually priorities for the Remix team to hash out (better i18n support is a big one). I figure others would benefit from knowing the general direction of where Remix is going. Thanks!

**Comment by cliffordfajardo** [comment]
Not ETA yet, but the Remix team did mention they are working on it; @chaance mentioned there yesterday https://twitter.com/chancethedev/status/1567958495185743872

**Reply by chaance** [reply]
Yep, this is a priority for me over the next couple of weeks. I'll share an update here when we've got something ready!

**Reply by osdiab** [reply]
i replied on the tweet, but overall i think a pinned discussion would do the job simply and be the easiest to discover :)

**Comment by chaance** [comment]
> better i18n support is a big one

@osdiab Care to elaborate here? What is it that we could do to make i18n easier for you?

**Reply by sergiodxa** [reply]
What's here https://github.com/remix-run/remix/discussions/2877 would be useful

**Reply by osdiab** [reply]
@chaance here are more relevant discussions, also cc @Xiphe who has been active on this subject:

- https://github.com/remix-run/remix/discussions/3355
- https://github.com/remix-run/remix/discussions/3847

**Reply by osdiab** [reply]
(and while i'm here, also better CSS handling is high on my list too - I spent the better part of a couple days trying and utterly failing at getting Emotion to work reliably with its `css` prop in Remix, and it's enough to drive me away from using Remix altogether)

- https://github.com/remix-run/remix/discussions/1302

**Reply by Xiphe** [reply]
Thanks for mentioning me Omar 👍 

I second all the above. To me the most relevant missing feature required for i18n support with great UX and DX is efficient loading of dynamic i18n strings.

From what I understand the most intuitive solution would be to support [dynamic blocking route dependencies](https://github.com/remix-run/remix/discussions/3355) and I see an opportunity to [elegantly provide this feature along with the integration of react router 6.4](https://github.com/remix-run/remix/discussions/3355#discussioncomment-3396705)

**Reply by osdiab** [reply]
Would be really cool if eventually Remix could work with https://github.com/ivanhofer/typesafe-i18n . Basically the DX that I would hope for and it's already built to be cross-platform. Hopefully the nav blocking task on the roadmap encapsulates the above core need :)

**Reply by sergiodxa** [reply]
> Would be really cool if eventually Remix could work with https://github.com/ivanhofer/typesafe-i18n . Basically the DX that I would hope for and it's already built to be cross-platform. Hopefully the nav blocking task on the roadmap encapsulates the above core need :)

Is there a reason why that doesn't work right now in Remix? Does it need compiler integration? Because I was able to make i18next work with Remix.

The i18n proposal I wrote is to build tools for i18n that needs router or compiler integration, but not attached to any particular library so you could use whatever you want.

**Comment by osdiab** [comment]
Stumbled upon what looks like a barebones roadmap here: https://github.com/orgs/remix-run/projects/5/views/1

**Reply by Xiphe** [reply]
Cool! Thanks for sharing

**Comment by ryanflorence** [comment]
Roadmap is here: https://github.com/orgs/remix-run/projects/5/views/1

We're in the process of making our development way more open. We were using Linear but are moving everything here to GitHub :)

**Comment by chaance** [comment]
I'm going to lock this conversation since we've got the roadmap now and folks should know where to find it. 🚀
